components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/tool_event_tab_management.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Icon from '@brave/leo/react/icon'
+import { formatLocale, getLocale } from '$web-common/locale'
+import type { Value } from 'gen/mojo/public/mojom/base/values.mojom.m.js'
+import { useUntrustedConversationContext } from '../../untrusted_conversation_context'
+import type { ToolComponent } from './tool_event'
+import styles from './tool_event_tab_management.module.scss'
+
+const ToolEventTabManagement: ToolComponent = (props) => {
+  const context = useUntrustedConversationContext()
+
+  const content = props.content
+
+  const needsUserPermission = props.toolUseEvent.requiresUserInteraction && props.isEntryActive
+
```

Comment:
needs a test

Reply:
looks like maybe the test isn't committed?